### PR TITLE
#63: Support for ScalaJs

### PR DESCRIPTION
**Pull request by FabioPinheiro**

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -32,13 +32,18 @@ commands += Command.command("ci-test") { s =>
     s
 }
 
-lazy val root = project
+// NOTE: We can also do this to run all matrix (crossScalaVersions * crossProject)
+addCommandAlias("ci-cross", "+ test; + publishLocal")
+
+lazy val root = crossProject(JVMPlatform, JSPlatform)
+  // .crossType(CrossType.Pure) // THIS IS NOT WORKING https://github.com/portable-scala/sbt-crossproject#configuration
+  // FIXME I add to create the 'shared' folder. Because we get erros like "not found: object zio"
```
Comment (FabioPinheiro) [inline build.sbt:40]: I'm not sure why this is not working on the project root ("." top level folder) already try a bunch of stuff

Is it ok to move `src` into the `shared` folder?

**Review by FabioPinheiro — COMMENTED** [review]
I will cleanup after

File: build.sbt [hunk 2]
```diff
@@ -20,25 +20,18 @@ val scala212 = "2.13.8"
...
-  s"++$scalaVersion" ::
-    "test" ::
-    "publishLocal" ::
-    s
-}
+// Run matrix (crossScalaVersions * crossProject)
+addCommandAlias("ci-cross", "+ test; + publishLocal")
 
-lazy val root = project
+lazy val root = crossProject(JVMPlatform, JSPlatform)
```
Comment (poslegm) [inline build.sbt:26]:
```suggestion
lazy val root = crossProject(JVMPlatform, JSPlatform)
  .withoutSuffixFor(JVMPlatform)
```

File: project/Dependencies.scala [hunk 3]
```diff
@@ -1,9 +1,16 @@
...
 object Dependencies {
-  private val munitVersion = "0.7.29"
-  private val zioVersion   = "2.0.1"
+  private val munitVersion  = "0.7.29"
+  private val zioVersion    = "2.0.1"
+  private val scalaJavaTime = "2.3.0"
 
-  lazy val munit = "org.scalameta" %% "munit" % munitVersion
-  lazy val zio   = "dev.zio"       %% "zio"   % zioVersion
+  lazy val munit = Def.setting("org.scalameta" %%% "munit" % munitVersion)
```
Comment (poslegm) [inline project/Dependencies.scala:9]: I think it's better to define dependencies directly in `build.sbt` then use `Def.settings`

File: build.sbt [hunk 4]
```diff
@@ -32,13 +32,18 @@ commands += Command.command("ci-test") { s =>
...
+  // .crossType(CrossType.Pure) // THIS IS NOT WORKING https://github.com/portable-scala/sbt-crossproject#configuration
+  // FIXME I add to create the 'shared' folder. Because we get erros like "not found: object zio"
   .in(file("."))
   .settings(
     name                                 := "munit-zio",
     libraryDependencies += "com.novocode" % "junit-interface" % "0.11" % "test",
     scalaVersion                         := scala3,
-    scalacOptions                        := Seq(
+    scalacOptions ++= Seq( // TODO Q: Is there any particular reason to use ":="?  That can break the plugin in unexpected ways
```
[the remark this answers is not in the document]
Comment (poslegm) [reply build.sbt:34]: I am OK with `++=`

File: build.sbt [hunk 5]
```diff
@@ -32,13 +32,18 @@ commands += Command.command("ci-test") { s =>
     s
 }
 
-lazy val root = project
+// NOTE: We can also do this to run all matrix (crossScalaVersions * crossProject)
+addCommandAlias("ci-cross", "+ test; + publishLocal")
```
[the remark this answers is not in the document]
Comment (poslegm) [reply build.sbt:24]: The reason for using `TEST` env is dedicated steps in CI job for Scala versions (see [example](https://github.com/poslegm/munit-zio/runs/7962880436?check_suite_focus=true)). But I agreed that it's a deprecated way. Please see how we can achieve dedicated steps more elegant way: [example](https://github.com/scalameta/scalameta/blob/main/.github/workflows/ci.yml#L13-L34).

File: build.sbt (hunk 1 above)
Comment (poslegm) [reply build.sbt:28]: FYI https://github.com/portable-scala/sbt-crossproject/issues/122

Please name this directory `core` (`core/src/main/....`).

[Build config example](https://github.com/typelevel/munit-cats-effect/blob/main/build.sbt).

**Review by poslegm — CHANGES_REQUESTED** [review]
Thank you a lot for PR!
A few small fixes and I am ready to ship it :)

File: build.sbt (hunk 1 above)
Comment (FabioPinheiro) [reply build.sbt:28]: I always have cross-project in subdirectories.
Didn't know about that limitation. Thanks

**Comment by FabioPinheiro** [comment]
I will do a clean up and force push squashed commit to this PR

**Comment by FabioPinheiro** [comment]
I'm going just to squash in case you want to do a simple merge commit

**Comment by poslegm** [comment]
LGTM! Let's go to merge. I will publish release in the evening.

**Review by poslegm — APPROVED** [review]

**Pull request merged**